frontend/test/metabase-lib/lib/Question.unit.spec.js

```diff
@@ -445,11 +472,35 @@ describe("Question", () => {
         expect(question.display()).toBe("scalar");
       });
 
-      it("should not set the display to scalar if table was selected", () => {
+      it("should not set the display to scalar if table was selected and display is locked", () => {
         const question = orders_count_question
           .setDisplay("table")
           .lockDisplay()
-          .maybeResetDisplay(["table", "scalar"]);
+          .maybeResetDisplay(ordersCountData, ["table", "scalar"]);
+
+        expect(question.display()).toBe("table");
+      });
+
+      it("should set the display to scalar if a non-scalar was selected and display is locked", () => {
+        const question = base_question
+          .setDisplay("table")
+          .maybeResetDisplay(ordersCountData, ["table", "scalar"]);
+
+        expect(question.display()).toBe("scalar");
+      });
+
+      it("should not set the display to scalar if another scalar display was selected and display is locked", () => {
+        const question = base_question
+          .setDisplay("gauge")
+          .maybeResetDisplay(ordersCountData, ["table", "scalar", "gauge"]);
+
+        expect(question.display()).toBe("gauge");
+      });
+
+      it("switch to table view if we had a scalar and now have more than 1x1 data", () => {
+        const question = base_question
+          .setDisplay("scalar")
+          .maybeResetDisplay(multipleRowsData, ["table"]);
 
         expect(question.display()).toBe("table");
       });
```

Comment:
I added this test because it demonstrates that the behaviour is the same after removing this conditional branch from the code
https://github.com/metabase/metabase/pull/33185/files#diff-fdff338c9e6c27877274c27c438055055f0f9395a0a37c3fd360e194f6d2e461L349-L350